code/python/example.py

```diff
@@ -1,21 +1,24 @@
-# pip3 install neo4j-driver
-# python3 example.py
+# (requires Python 3.6 or newer and pip, check with `pip --version`)
+# pip install neo4j
+# python example.py
 
-from neo4j import GraphDatabase, basic_auth
+from neo4j import GraphDatabase
 
-driver = GraphDatabase.driver(
-  "neo4j+s://demo.neo4jlabs.com:7687",
-  auth=basic_auth("mUser", "s3cr3t"))
+def products_of_category(transaction, movie_title_):
+    cypher_query = (
+        "MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person)" 
+        "RETURN a.name as actorName")
+    result = transaction.run(cypher_query, movie_title=movie_title_)
+    return result.value("actorName")
 
-cypher_query = '''
-MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person) RETURN a.name as actorName
-'''
+uri = "neo4j+s://demo.neo4jlabs.com:7687"
+auth = ("mUser", "s3cr3t")
+driver = GraphDatabase.driver(uri, auth=auth)
 
 with driver.session(database="movies") as session:
-  results = session.read_transaction(
-    lambda tx: tx.run(cypher_query,
-                      movieTitle="The Matrix").data())
-  for record in results:
-    print(record['actorName'])
+    categories = session.read_transaction(movies,
+                                          "The Matrix")
+    for category in categories:
+        print(category)
 
-driver.close()
+driver.close()
```

Comment:
can driver not be auto-closes with a `with`?

Reply:
Just saw, that it is actually documented, just not very prominently. So I'm happy to put it in.